serverless/pytorch/mmpose/ubody2d/nuclio/main.py

```diff
@@ -0,0 +1,72 @@
+import json
+import base64
+import io
+import yaml
+import numpy as np
+from PIL import Image
+
+from mmpose.apis import MMPoseInferencer
+
+def init_context(context):
+    context.logger.info("Init detector...")
+
+    det_config = "/opt/nuclio/mmpose/projects/rtmpose/rtmdet/person/rtmdet_nano_320-8xb32_coco-person.py"
+    det_checkpoint = "/opt/nuclio/rtmdet_nano_8xb32-100e_coco-obj365-person-05d8511e.pth"
+    pose_config = "/opt/nuclio/mmpose/configs/wholebody_2d_keypoint/topdown_heatmap/ubody2d/td-hm_hrnet-w32_8xb64-210e_ubody-256x192.py"
+    pose_checkpoint = "/opt/nuclio/td-hm_hrnet-w32_8xb64-210e_ubody-coco-256x192-7c227391_20230807.pth"
+
+    inferencer = MMPoseInferencer(
+        pose2d=pose_config,
+        pose2d_weights=pose_checkpoint,
+        det_model=det_config,
+        det_weights=det_checkpoint,
+        det_cat_ids=[0],  # the category id of 'human' class
+        device='cpu'
+    )
+
+    context.logger.info("Init labels...")
+    with open("/opt/nuclio/function.yaml", "rb") as function_file:
+        functionconfig = yaml.safe_load(function_file)
+        labels_spec = functionconfig["metadata"]["annotations"]["spec"]
+        labels = [{
+            "name": item["name"],
+            "elements": item["elements"]
+        } for item in json.loads(labels_spec)]
+
+    context.user_data.labels = labels
+    context.user_data.inferencer = inferencer
+    context.logger.info("Function initialized")
+
+def handler(context, event):
+    context.logger.info("Run mmpose ubody-2d model")
+    data = event.body
+    buf = io.BytesIO(base64.b64decode(data["image"]))
+    image = Image.open(buf).convert("RGB")
+
+    results = []
+    pred_instances = next(context.user_data.inferencer(np.array(image)[...,::-1]))["predictions"][0]
+
+    for pred_instance in pred_instances:
+        keypoints = pred_instance["keypoints"]
+        keypoint_scores = pred_instance["keypoint_scores"]
+        for label in context.user_data.labels:
+            skeleton = {
+                "confidence": str(pred_instance["bbox_score"]),
+                "label": label["name"],
+                "type": "skeleton",
+                "elements": [{
+                    "label": element["name"],
+                    "type": "points",
+                    "outside": 0 if 0.66 < keypoint_scores[element["id"]] < 0.98 else 1,
```

Comment:
Why 0.66 < keypoint_scores[element["id"]] < 0.98?
Why not 0.66 < keypoint_scores[element["id"]]?
Why the low boundary is 0.66? Look at https://github.com/opencv/cvat/blob/ba3db011f848b4ba005ea08d2c17527f52b8a955/serverless/tensorflow/matterport/mask_rcnn/nuclio/main.py#L27. Can we use something similar?

Reply:
@nmanovic 

As I told above, these values were found experementally. I don't know exactly what this confidence means in this specific model, but it is too low, or too high, there are a lot of false positives

Reply:
Their demo uses just > 0.3, but with this condition it does not work well